Revise style checker guidelines for RefPtr variable names

abigailfox opened the pull request:
#### 343bda4006d6ec3dc4e3ff344eec463be8127cee
<pre>
Revise style checker guidelines for RefPtr variable names
<a href="https://bugs.webkit.org/show_bug.cgi?id=272485">https://bugs.webkit.org/show_bug.cgi?id=272485</a>
<a href="https://rdar.apple.com/126233155">rdar://126233155</a>

Reviewed by Chris Dumez and David Kilzer.

Our current style guidelines discourage the use of the &quot;protected&quot; prefix or &quot;protector&quot; in new variable names when creating ref counted variables.
This updates the style checker to align with the current standard.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_identifier_name_in_declaration):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(WebKitStyleTest.test_names):

Canonical link: <a href="https://commits.webkit.org/277343@main">https://commits.webkit.org/277343@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5302dbb432ac5dae14736dcdc008abfafc50deb6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47375 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26555 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50027 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50058 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43423 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/49682 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32073 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24015 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/38571 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47956 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24092 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40821 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19891 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/47238 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/21530 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41987 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5418 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/43722 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42395 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51934 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22405 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/18742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/45869 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23680 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44900 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10443 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24466 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23399 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
```diff
@@ -6276,66 +6276,71 @@ def test_names(self):
+        self.assert_lint('Ref<Node> nodeRef = node;', '')
 
         # Invalid protector RefPtr/Ref variable names.
+        self.assert_lint('RefPtr<Node> protectedNode(&node);', "'protectedNode' is incorrectly named. New variables should not use the 'protected' prefix.  [readability/naming/protected] [4]")
```
Super-nit:  This test case from original line 6279 was not converted to a failing case (there were 20 removed test cases in the first section above, but only 19 added as invalid test cases below):

```
        self.assert_lint('RefPtr<Node> protectedNode(node);', '')
```

ddkilzer reviewed (approved):
r=me, but as an enforcer of the proper use of `protect(ed|or)` variables, @cdumez should probably review, too. :)

cdumez reviewed (approved):
Seems OK

abigailfox committed:
Revise style checker guidelines for RefPtr variable names
https://bugs.webkit.org/show_bug.cgi?id=272485
rdar://126233155

Reviewed by Chris Dumez and David Kilzer.

Our current style guidelines discourage the use of the "protected" prefix or "protector" in new variable names when creating ref counted variables.
This updates the style checker to align with the current standard.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_identifier_name_in_declaration):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(WebKitStyleTest.test_names):

Canonical link: https://commits.webkit.org/277343@main
